judgels-backends/uriel/uriel-app/src/main/java/judgels/uriel/contest/ContestStore.java

```diff
@@ -176,6 +212,94 @@ public Contest createContest(ContestCreateData contestCreateData) {
+        contestModel.slug = contestDump.getSlug();
+        contestModel.name = contestDump.getName();
+        contestModel.style = contestDump.getStyle().getName().name();
+        contestModel.beginTime = contestDump.getBeginTime();
+        contestModel.duration = contestDump.getDuration().toMillis();
+        contestModel.description = contestDump.getDescription();
+        contestDao.setModelMetadataFromDump(contestModel, contestDump);
+        contestModel = contestDao.persist(contestModel);
+        contestByJidCache.invalidate(contestModel.jid);
+        contestBySlugCache.invalidate(contestModel.slug);
+
+        String contestJid = contestModel.jid;
+        contestModuleStore.importStyleDump(contestJid, contestDump.getStyle());
+
+        contestDump.getModules().forEach(dump -> contestModuleStore.importModuleDump(contestJid, dump));
+        contestDump.getProblems().forEach(dump -> contestProblemStore.importDump(contestJid, dump));
+        contestDump.getContestants().forEach(dump -> contestContestantStore.importDump(contestJid, dump));
+        contestDump.getSupervisors().forEach(dump -> contestSupervisorStore.importDump(contestJid, dump));
+        contestDump.getManagers().forEach(dump -> contestManagerStore.importDump(contestJid, dump));
+        contestDump.getAnnouncements().forEach(dump -> contestAnnouncementStore.importDump(contestJid, dump));
+        contestDump.getClarifications().forEach(dump -> contestClarificationStore.importDump(contestJid, dump));
+
+        return contestJid;
+    }
+
+    public Set<ContestDump> exportDumps(Map<String, ExportContestsDumpData.ContestDumpEntry> contestsToExport) {
+        Set<String> contestJidsToExport = contestsToExport.keySet();
+
+        return contestDao.selectByJids(contestJidsToExport).values().stream()
+                .map(contestModel -> {
+                    DumpImportMode mode = contestsToExport.get(contestModel.jid).getMode();
+                    Set<ContestDumpComponent> components = contestsToExport.get(contestModel.jid).getComponents();
+
+                    ContestDump.Builder builder = new ContestDump.Builder()
+                            .mode(mode)
+                            .slug(contestModel.slug)
+                            .name(contestModel.name)
+                            .beginTime(contestModel.beginTime)
+                            .duration(Duration.ofMillis(contestModel.duration))
+                            .description(contestModel.description)
+                            .style(contestModuleStore.exportStyleDump(
+                                    contestModel.jid, mode, ContestStyle.valueOf(contestModel.style)))
+                            .modules(contestModuleStore.exportModuleDumps(contestModel.jid, mode));
+
+                    if (components.contains(ContestDumpComponent.PROBLEMS)) {
+                        builder = builder.problems(contestProblemStore.exportDumps(contestModel.jid, mode));
```

Comment:
these assignment are not needed; the methods are mutating the builder already

(in all places)

Reply:
Changed